pkg/store/mongodb/mongodbStore.go

```diff
@@ -0,0 +1,852 @@
+
+	opts := options.Aggregate().SetMaxTime(5 * time.Second)
+	cur, err := serviceCollection.Aggregate(ctx, mongo.Pipeline{matchstag, projectstag}, opts)
+	if err != nil {
+		return -1, err
+	}
+	var results []bson.M
+	if err = cur.All(ctx, &results); err != nil {
+		panic(err)
+	}
+	if results == nil || len(results) == 0 {
+		if len(serviceName) != 0 {
+			return -1, errors.Errorf(errors.EntityNotFound, "service %q is not found", serviceName)
+		} else {
+			return 0, nil
+		}
+	}
+	policyCount := int64(0)
+
+	for _, res := range results {
+		policyCount += int64(res["policycount"].(int32))
+	}
+
+	return policyCount, nil
+
+}
+
+func (s *Store) GetPolicy(serviceName string, id string) (*pms.Policy, error) {
+	serviceCollection := s.client.Database(s.Database).Collection("services")
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	matchstag := bson.D{{"$match", bson.D{{"_id", serviceName}}}}
+	projectstag := bson.D{
+		{"$project", bson.D{
+			{"policies", bson.D{
+				{"$filter", bson.D{
+					{"input", "$policies"},
+					{"as", "p"},
+					{"cond", bson.D{{"$eq", bson.A{"$$p._id", id}}}}},
+				}},
+			}},
+		}}
+	opts := options.Aggregate().SetMaxTime(5 * time.Second)
+	cur, err := serviceCollection.Aggregate(ctx, mongo.Pipeline{matchstag, projectstag}, opts)
+	if err != nil {
+		return nil, err
+	}
+	var services []pms.Service
+	if err = cur.All(ctx, &services); err != nil {
+		panic(err)
+	}
+	if services == nil || len(services) == 0 {
+		return nil, errors.Errorf(errors.EntityNotFound, "service %q is not found", serviceName)
+	}
+	if services[0].Policies == nil || len(services[0].Policies) == 0 {
+		return nil, errors.Errorf(errors.EntityNotFound, "policy %q is not found", id)
+	}
+	return services[0].Policies[0], nil
+
+}
+
+func (s *Store) DeletePolicy(serviceName string, id string) error {
+	serviceCollection := s.client.Database(s.Database).Collection("services")
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	filter := bson.D{{"_id", serviceName}}
+	update := bson.D{{"$pull", bson.D{{"policies", bson.D{{"_id", id}}}}}}
+	result, err := serviceCollection.UpdateOne(ctx, filter, update)
+	if err != nil {
+		return err
+	}
+	if result.MatchedCount == 0 {
+		return errors.Errorf(errors.EntityNotFound, "service %q is not found", serviceName)
+	}
+	if result.ModifiedCount == 0 {
+		return errors.Errorf(errors.EntityNotFound, "policy %q is not found", id)
+	}
+	return nil
+
+}
+
+func (s *Store) DeletePolicies(serviceName string) error {
+	serviceCollection := s.client.Database(s.Database).Collection("services")
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	filter := bson.D{{"_id", serviceName}}
+	update := bson.D{{"$pull", bson.D{{"policies", bson.D{{"$exists", true}}}}}}
+	result := serviceCollection.FindOneAndUpdate(ctx, filter, update)
+	if result.Err() == mongo.ErrNoDocuments {
```

Comment:
I believe the segment should be:
```
if result.Err() != nil {
    if result.Err() == mongo.ErrNoDocuments {
         return errors.Errorf(xxxxx)
    }
    return result.Err()
}
return nil
```

Reply:
It seems no need to check if result.Err() != nil  here.